ui/perfherder/helpers.js

```diff
@@ -429,25 +429,53 @@ export const addResultsLink = (taskId) => {
   return `${taskLink}${taskId}${resultsPath}`;
 };
 
-export const getTextualSummary = (
-  alerts,
-  alertSummary,
-  copySummary = null,
-  alertsWithVideos = [],
-) => {
-  let resultStr = '';
-  const improved = sortBy(
-    alerts.filter((alert) => !alert.is_regression),
-    'amount_pct',
-  ).reverse();
-  const regressed = sortBy(
-    alerts.filter(
-      (alert) => alert.is_regression && alert.status !== alertStatusMap.invalid,
-    ),
-    'amount_pct',
-  ).reverse();
+export class TextualSummary {
+  constructor(alerts, alertSummary, copySummary = null, alertsWithVideos = []) {
+    this.alerts = alerts;
+    this.alertSummary = alertSummary;
+    this.copySummary = copySummary;
+    this.alertsWithVideos = alertsWithVideos;
+    this.resultStr = '';
+  }
+
+  get markdown() {
+    const improved = sortBy(
+      this.alerts.filter((alert) => !alert.is_regression),
+      'amount_pct',
+    ).reverse();
+    const regressed = sortBy(
+      this.alerts.filter(
+        (alert) =>
+          alert.is_regression && alert.status !== alertStatusMap.invalid,
+      ),
+      'amount_pct',
+    ).reverse();
+
+    const regressionsTable = this.getFormattedRegressions(regressed);
+    this.resultStr += regressionsTable;
+    const improvementsTable = this.getFormattedImprovements(improved);
+    this.resultStr += improvementsTable;
+
+    if (this.copySummary) {
+      this.attachSummaryHeaderAndAlertLink();
```

Comment:
Even if you're changing some variables inside this function, returning them back at client level makes the code easier to read.